Bug 1145112 - Convert all usages of `wait_for_condition` to `Wait().until()`. r=whimboo

sr-murthy opened the pull request:
Replaced all instances of `wait_for_condition()` with `Wait().until()`

whimboo commented on firefox_ui_tests/functional/keyboard_shortcuts/test_browser_window.py:
```diff
@@ -50,4 +50,4 @@ def has_input_selected(mn):
             return selection_name == "input"
 
-        self.wait_for_condition(has_input_selected)
+        Wait(self.marionette).until(has_input_selected)
```
So this change looks fine, but have you run the tests locally yourself? At least this test will fail because `Wait` has not been imported.

whimboo commented on firefox_ui_tests/functional/locationbar/test_access_locationbar.py:
```diff
@@ -45,13 +45,15 @@ def load_urls():
-        self.wait_for_condition(lambda _: len(self.autocomplete_results.visible_results) > 1)
+        Wait(self.marionette).until(lambda _: self.autocomplete_results.is_open)
+        Wait(self.marionette).until(
+            lambda _: len(self.autocomplete_results.visible_results) > 1))
```
I would propose that you combine the conditions in a single `until` call.

whimboo commented on firefox_ui_tests/functional/locationbar/test_suggest_bookmarks.py:
```diff
@@ -72,15 +72,16 @@ def visit_urls():
+        Wait(self.marionette).until(lambda mn: locationbar.value == search_string)
+        Wait(self.marionette).until(lambda mn: autocomplete_results.is_open)
+        Wait(self.marionette).until(
+            lambda mn: len(autocomplete_results.visible_results) == 1)
```
Here I would also combine the autocomplete conditions into a single `until` call.

whimboo commented on firefox_ui_tests/remote/security/test_safe_browsing_notification.py:
```diff
@@ -83,9 +83,10 @@ def check_ignore_warning_button(self, unsafe_page):
+        Wait(self.marionette).until(expected.element_stale(button))
+        Wait(self.marionette).until(expected.element_present(By.ID, 'main-feature'))
+        self.assertEquals(self.marionette.get_url(),\
+        self.browser.get_final_url(unsafe_page))
```
Please keep the assert line as it was before. The wrapping is not necessary.

whimboo commented:
Beside the above comments please also check the reported Travis failures and get them fixed. Travis CI has to pass before we can merge anything.

sr-murthy commented:
OK, will do these soon.

sr-murthy commented:
I've made the requested changes, but I cannot acces the Travis CI build.  Also the build did not trigger in the normal way.

whimboo commented:
Thanks for the update! So the test via Travis has not been triggered because there is a merge conflict. You will have to rebase your development branch against the latest changes on upstream/master first. Once this is done Travis will pick up the latest state of the branch and run the tests.

sr-murthy commented:
I thought I had fixed the rebase conflicts, since I was able to finish the rebase.  Now the Travis build fails again.  I really don't understand what I'm doing wrong.

sr-murthy commented:
Looking at the build failure the errors seem related to PEP8 checks, not my code changes.

whimboo commented:
It looks like that you got the merge done now. That's great. But as you can see Travis is still not happy due to Pep8 problems. You will have to go through each of the lines and fix what the test output shows. If you have specific questions you can always join us on IRC and we can have a direct conversation about a specific problem.

sr-murthy commented:
OK, I'll turn off the Travis CI build option for the fork and then push again.

sr-murthy commented:
Sorry for the delay, but I in a week I will fix the PEP8 errors and update the pull request.

sr-murthy commented:
RIght, finally I have been able to find the time to resolve the PEP8 errors.  I note that your advice to combine two `Wait().until()` statements into one conflicts with the PEP8 line length limit of 99 characters per line.  This is the error I get for line 52 in `functional/locationbar/test_access_locationbar.py` when I run the `pep8` tool locally: 

    $ pep8 --max-line-length=99 --exclude=client firefox_ui_tests/functional/locationbar/test_access_locationbar.py
    firefox_ui_tests/functional/locationbar/test_access_locationbar.py:51:100: E501 line too long (136 > 99 characters)

sr-murthy commented:
RIght, finally I have been able to find the time to resolve the PEP8 errors.  I note that your advice to combine two `Wait().until()` statements into one conflicts with the PEP8 line length limit of 99 characters per line.  This is the error I get for line 52 in `functional/locationbar/test_access_locationbar.py` when I run the `pep8` tool locally: 

    $ pep8 --max-line-length=99 --exclude=client firefox_ui_tests/functional/locationbar/test_access_locationbar.py
    firefox_ui_tests/functional/locationbar/test_access_locationbar.py:51:100: E501 line too long (136 > 99 characters)

sr-murthy commented:
RIght, finally I have been able to find the time to resolve the PEP8 errors.  I note that your advice to combine two `Wait().until()` statements into one conflicts with the PEP8 line length limit of 99 characters per line.  This is the error I get for line 52 in `functional/locationbar/test_access_locationbar.py` when I run the `pep8` tool locally: 

    $ pep8 --max-line-length=99 --exclude=client firefox_ui_tests/functional/locationbar/test_access_locationbar.py
    firefox_ui_tests/functional/locationbar/test_access_locationbar.py:51:100: E501 line too long (136 > 99 characters)

sr-murthy commented:
I've resolved the PEP8 errors (and testing this locally using the `pep8` tool) but again rebasing the development branch against my updated local master causes problems.  Specifically, what I do is do a `git fetch` of the upstream master to the local master then `merge`, switch to the dev branch and do a `git rebase -i master`.  I choose only the commits I want, deleting those which are unnecessary, and squashing all the most recent relevant commits to the first one, but I get the following message:

    git rebase -i master
    error: could not apply 1a712f6... Replaced all instances of wait_for_condition() with Wait().until()

    When you have resolved this problem, run "git rebase --continue".
    If you prefer to skip this patch, run "git rebase --skip" instead.
    To check out the original branch and stop rebasing, run "git rebase --abort".
    Could not apply 1a712f6cb827e631d906138ac81d6fbfd10e0349... Replaced all instances of wait_for_condition() with Wait().until()

sr-murthy commented:
I'm going to close this, and create a new local dev branch from which I will create a new pull request.
